src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -317,14 +317,22 @@ std::unique_ptr<model_backend> tf_compute_context::create_activity_classifier(
   const float* pred_window = prediction_window.data();
   int pw = static_cast<int>(*pred_window);
 
+  int seed = -1;
+  auto random_seed_it = config.find("random_seed");
+  if (random_seed_it != config.end()) {
+    shared_float_array random_seed = random_seed_it->second;
+    const float* rd = random_seed.data();
+    seed = static_cast<int>(*rd);
```

Comment:
Using static cast here loses precision. See how this field is set here:
https://github.com/apple/turicreate/blob/76618e56d00c7db542a47a301e9c4b6f39ee883e/src/toolkits/activity_classification/activity_classifier.cpp#L65

And how the MPS implementation reads it:
https://github.com/apple/turicreate/blob/76618e56d00c7db542a47a301e9c4b6f39ee883e/src/ml/neural_net/mps_layers.mm#L655

In general, using on the "integer" part of the `float` value won't use the full 32 bits of entropy we allow to be passed in the random seed

Reply:
Okay read the difference makes a bit more sense.